packages/website/docs/using-react.md

```diff
@@ -0,0 +1,179 @@
+
+## Creating the component
+
+Start with some boilerplate for creating a React component. This component uses the [`useRef`](https://reactjs.org/docs/hooks-reference.html#useref) hook to create a mutable ref object, `containerRef`, to mount the autocomplete on. To learn more about this hook, check out the [`useRef` React documentation](https://reactjs.org/docs/hooks-reference.html#useref).
+
+ All that you need to render is a `div` with the `containerRef` as the `ref`.
+
+```jsx title="Autocomplete.jsx"
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+### Mounting the autocomplete
+
+Now that you have access to the DOM through the `containerRef` object, you can create and mount the Autocomplete instance. Upon instantiation, you can include any desired [Autocomplete options](basic-options) and rely on `props` to pass any options you want to remain configurable.
+
+The example component below sets only the [`container`](autocomplete-js/#container) option. It specifies where to mount your Autocomplete component, but lets all other [options](basic-options) get configured through props.
+
+**You must also pass the `renderer` and `render` parameters.** This is because the default Autocomplete implementation uses [Preact's](https://preactjs.com/) version of `createElement`, `Fragment` and `render`. Without providing React's version of these, the Autocomplete instance won't render the views properly.
+
+Don't forget to [clean up the effect](https://reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect) by returning a function that destroys the Autocomplete instance.
+
+```jsx title="Autocomplete.jsx"
+import { autocomplete } from '@algolia/autocomplete-js';
+import React, { createElement, Fragment, useEffect, useRef } from 'react';
+import { render } from 'react-dom';
+
+export function Autocomplete(props) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (!containerRef.current) {
+      return undefined;
+    }
+
+    const search = autocomplete({
+      container: containerRef.current,
+      renderer: { createElement, Fragment },
+      render({ children }, root) {
+        render(children, root);
+      },
+      ...props,
+    });
+
+    return () => {
+      search.destroy();
+    };
+  }, [props]);
+
+  return (
+    <div
+      ref={containerRef}
+    />
+  );
+}
+```
+
+## Using the component
+
+Now that you've created an `<Autocomplete/>` component, you can use it in your React application.
+
+As previously state, the example component sets only the [`container`](autocomplete-js/#container) option, but lets all other [options](basic-options) get configured through props.
```

Comment:
Is this paragraph needed as it was already said above?

Reply:
True -- https://github.com/algolia/autocomplete.js/pull/442/commits/8c6843808f31dcaa6878e49af294304501700df2